### PR TITLE
Refactor AccumulateCredit

### PR DESCRIPTION
**Pull request by taikiy**
* Make `AccumulateCredit` struct signatures align with other protocols.
* Remove `IterStep`
* Rename values/structs

### CODE REVIEW

File: src/protocol/attribution/accumulate_credit.rs [hunk 1]
```diff
@@ -54,28 +51,26 @@ impl<'a, F: Field> AccumulateCredit<'a, F> {
...
             .collect::<Vec<_>>()
             .try_into()
             .unwrap();
 
-        let mut credits: Batch<Replicated<F>> = self
-            .input
+        let mut credits: Batch<Replicated<F>> = input
             .iter()
-            .map(|x| x.value.clone())
+            .map(|x| x.credit.clone())
```
Comment (taikiy) [inline src/protocol/attribution/accumulate_credit.rs:73]: The logic is unchanged. Just renaming the field to align with other attribution protocols to be added.

File: src/protocol/attribution/accumulate_credit.rs [hunk 2]
```diff
@@ -94,38 +89,38 @@ impl<'a, F: Field> AccumulateCredit<'a, F> {
         // iteration, and the interaction do not depend on the calculation results
         // of other elements, allowing the algorithm to be executed in parallel.
 
-        let mut iteration_step = IterStep::new("iteration", 0);
-
         // generate powers of 2 that fit into input len. If num_rows is 15, this will produce [1, 2, 4, 8]
-        for step_size in std::iter::successors(Some(1u32), |prev| prev.checked_mul(2))
+        for (depth, step_size) in std::iter::successors(Some(1u32), |prev| prev.checked_mul(2))
```
Comment (taikiy) [inline src/protocol/attribution/accumulate_credit.rs:93]: `depth` is log N. For our case, 32 should be enough.

File: src/protocol/attribution/accumulate_credit.rs [hunk 3]
```diff
@@ -268,34 +272,32 @@ mod tests {
...
+            [1, 1, 0, 12],
+            [0, 1, 2, 0],
+            [0, 1, 2, 0],
+            [1, 1, 0, 6],
+            [1, 1, 0, 4],
+            [0, 1, 5, 0],
+            [1, 1, 5, 6],
+        ];
+        const EXPECTED: &[u128] = &[
+            0, 0, 19, 19, 9, 7, 6, 1, 0, 10, 15, 15, 12, 0, 10, 10, 4, 6, 6,
```
Comment (taikiy) [inline src/protocol/attribution/accumulate_credit.rs:286]: Adding more events to catch some corner case bugs. This is quite complicated protocol/code so we need a better set of test cases.

File: src/protocol/attribution/accumulate_credit.rs [hunk 4]
```diff
@@ -28,24 +31,18 @@ impl AsRef<str> for Step {
...
-}
-
-impl<'a, F: Field> AccumulateCredit<'a, F> {
-    #[allow(dead_code)]
-    pub fn new(input: &'a Batch<AttributionInputRow<F>>) -> Self {
-        Self { input }
-    }
+pub struct AccumulateCredit {}
 
+impl AccumulateCredit {
```
Comment (benjaminsavage) [inline src/protocol/attribution/accumulate_credit.rs:45]: Honestly, we are eliminating the use of structs for all the other protocols. We should probably just eliminate the struct entirely in this PR.

File: src/protocol/attribution/accumulate_credit.rs [hunk 5]
```diff
@@ -208,17 +206,22 @@ impl<'a, F: Field> AccumulateCredit<'a, F> {
...
+        },
         test_fixture::{share, Reconstruct, TestWorld},
     };
     use rand::rngs::mock::StepRng;
     use std::iter::zip;
     use tokio::try_join;
 
+    /// Takes a vector of 4-element vectors (e.g., `RAW_INPUT`), and create
+    /// shares of `AttributionInputRow`.
+    // TODO(taikiy): Implement a `InfoShares` for any struct
```
Comment (benjaminsavage) [inline src/protocol/attribution/accumulate_credit.rs:224]: Do you mean `IntoShares`?

File: src/protocol/attribution/accumulate_credit.rs (hunk 3 above)
Comment (benjaminsavage) [inline src/protocol/attribution/accumulate_credit.rs:286]: This is extremely hard to compare to the input. Would be cool if they were on the same line.

**Review by benjaminsavage — APPROVED** [review]
This looks great. The step approach is *much* better. Thank you for splitting things up!
